### PR TITLE
Add CSI500 Featuer (Surport CSI500 Data)

### PR DESCRIPTION
**Pull request by GaochaoZhu**
## How Has This Been Tested?
- [x] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [x] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:
`python qlib\scripts\data_colector\cn_index\collector.py --index_name CSI500 --qlib_dir
 ~/.qlib/qlib_data/cn_data --method parse_instruments `

    `2020-12-26 18:50:36.589 | INFO     | data_collector.utils:get_calendar_list:68 - end of get calendar list: CSI500.
100%█████████████████████████████████████████████████| 700/700 [00:35<00:00, 19.57it/s]`

    `2020-12-26 18:50:49.202 | INFO     | data_collector.index:parse_instruments:207 - parse csi500 companies finished.`
## Types of changes

- [x] Fix bugs
- [x] Add new feature
- [x] Update documentation

### CODE REVIEW

**Comment by you-n-g** [comment]
@GaochaoZhu   Thanks so much!  :) 
@zhupr  will help to review this PR later.

File: scripts/data_collector/cn_index/collector.py [hunk 1]
```diff
@@ -21,21 +23,21 @@
...
 INDEX_CHANGES_URL = "http://www.csindex.com.cn/zh-CN/search/total?key=%E5%85%B3%E4%BA%8E%E8%B0%83%E6%95%B4%E6%B2%AA%E6%B7%B1300%E5%92%8C%E4%B8%AD%E8%AF%81%E9%A6%99%E6%B8%AF100%E7%AD%89%E6%8C%87%E6%95%B0%E6%A0%B7%E6%9C%AC%E8%82%A1%E7%9A%84%E5%85%AC%E5%91%8A"
 
 
 class CSIIndex(IndexBase):
+
     @property
     def calendar_list(self) -> List[pd.Timestamp]:
         """get history trading date
 
-        Returns
```
Comment (zhupr) [inline scripts/data_collector/cn_index/collector.py:35]:
```python
def fun(a, b):
    """
    """
```
numpy style: https://numpydoc.readthedocs.io/en/latest/format.html

File: scripts/data_collector/cn_index/collector.py [hunk 2]
```diff
@@ -152,7 +154,9 @@ def _read_change_from_url(self, url: str) -> pd.DataFrame:
             df_map = pd.read_excel(_io, sheet_name=None)
             with self.cache_dir.joinpath(
                 f"{self.index_name.lower()}_changes_{add_date.strftime('%Y%m%d')}.{excel_url.split('.')[-1]}"
-            ).open("wb") as fp:
+            ).open(
```
Comment (zhupr) [inline scripts/data_collector/cn_index/collector.py:157]: Formatting codes using black:
```shell
# pip install  black
black <python file or dir> -l 120
```

**Comment by you-n-g** [comment]
This feature is merged in https://github.com/microsoft/qlib/pull/938

Welcome Gaochao become one of [Qlib's public contributors](https://github.com/microsoft/qlib/graphs/contributors)
Thanks to your previous contribution which is not publicly listed as well! 

Looking forward to your future contribution!!! :)

**Comment by you-n-g** [comment]
@GaochaoZhu 
This feature is merged in https://github.com/microsoft/qlib/pull/938

Welcome Gaochao become one of [Qlib's public contributors](https://github.com/microsoft/qlib/graphs/contributors)
Thanks to your previous contribution which is not publicly listed as well! 

Looking forward to your future contribution!!! :)

**Pull request closed without merging**